shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ingest/MySQLInventoryDumper.java

```diff
@@ -42,7 +44,9 @@ public MySQLInventoryDumper(final InventoryDumperConfiguration inventoryDumperCo
     
     @Override
     public Object readValue(final ResultSet resultSet, final int index) throws SQLException {
-        if (isDateTimeValue(resultSet.getMetaData().getColumnType(index))) {
+        if (isYearDataType(resultSet.getMetaData().getColumnTypeName(index))) {
+            return resultSet.getObject(index) == null ? null : resultSet.getShort(index);
```

Comment:
Could `resultSet.getObject(index) == null ? null : resultSet.getShort(index)` be simplified to `resultSet.getShort(index)` and `resultSet.wasNull()`?

If not, `resultSet.getObject(index) == null` should be `null == resultSet.getObject(index)`.